added java annotations by referring Order.hbm.xml

ajeyprasand opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'TRUNK-5111: Replace use of deprecated isVoided' -->
<!--- 'TRUNK-JiraIssueNumber: JiraIssueTitle' -->
## Description of what I changed
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first! -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it! -->
<!--- Just add the issue number at the end: -->
see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

ManojLL reviewed (commented):
@ajeyprasand Thanks for the PR, this implantation is not completed you have to remove existing xml mapping and update test.
please refer this PRs: https://github.com/openmrs/openmrs-core/commit/335c2b5186086f8a3b3d1d353bb1d06d23a20dcd

https://github.com/openmrs/openmrs-core/commit/b2bd587e6a92c5b328a92efc43ce5dcd6f4ea594

ajeyprasand commented:
@ManojLL I have updated the changes as requested. Do check and share your feedback

ManojLL commented:
@ajeyprasand some changes are  not directly related to the scope of this JIRA ticket `TRUNK-5881`. when creating a PR that should focus only about related changes, in this case `DrugOrder, ReferalOrder..` entities are having separate tickets. you can open separate PRs for those.

ajeyprasand commented:
oh but in the order hibernate xml file, it had configurations for drug order, referral order, test order. and I cant find tickets for these too.

rishabhrawat05 reviewed (commented):
You have to create annotations for the Order Domain only and keep the rest as it is. These changes should include creating annotations for the Order Domain, deleting the Order hibernate config file, removing the hibernate mapping, and adding a test for the Order Domain.

ajeyprasand commented:
Oh, okay. I got a bit confused because there was no hibernate xml file for DrugOrder, etc and the hibernate xml file for Order had the configurations for them. So, I initially thought that deleting the Order XML file might result in data loss, which is why I went ahead and added Hibernate annotations for them. But now, I've removed those annotations. Please take a look and let me know @rishabhrawat05  @ManojLL

rishabhrawat05 commented on api/src/test/java/org/openmrs/OrderTest.java:
```diff
@@ -58,7 +58,7 @@ public void setUp() throws Exception {
 	}
 
-	
+
```
Please remove this line

rishabhrawat05 commented on api/src/test/java/org/openmrs/api/OrderServiceTest.java:
```diff
@@ -60,6 +60,7 @@
 import org.openmrs.ProgramAttributeType;
 import org.openmrs.Provider;
 import org.openmrs.ProviderAttributeType;
+import org.openmrs.ReferralOrder;
```
Please remove this import statement of ReferralOrder

rishabhrawat05 commented on api/src/main/java/org/openmrs/ServiceOrder.java:
```diff
@@ -9,12 +9,15 @@
 package org.openmrs;
 
+import javax.persistence.MappedSuperclass;
+
```
Please remove all the changes from this file

ajeyprasand replied:
@rishabhrawat05  do I also have to add getter and setter for attributes which is a field newly added, and include it in copyHelper and cloneForRevision?

rishabhrawat05 replied:
> @rishabhrawat05 do I also have to add getter and setter for "attributes" of type Set which is a field newly added, and include it in copyHelper and cloneForRevision?

No, I think there is no need for that

ajeyprasand replied:
@rishabhrawat05 okk. Do review the pr now.

rishabhrawat05 reviewed (approved):
@ManojLL can you review this

ajeyprasand commented:
@ManojLL @rishabhrawat05  According to the logs, in org.openmrs.MedicationDispense, there is an @ManyToOne  in drugorder, but since drugOrder is not an entity its throwing an error. Drug order configuration was mentioned in the order hibernate xml file. Now since its deleted, its throwing the error.

ajeyprasand commented:
After reverting to the changes I made during "added required changes" commit its working

ajeyprasand commented:
@ManojLL @rishabhrawat05  but If I revert the changes to "added required changes" commit and modify a little more, its working fine

dkayiwa commented:
@ajeyprasand did you see the merge conflicts?

ajeyprasand commented:
TRUNK-5881: Order Domain - Switching from Hibernate Mappings to Annotations #5048
@ManojLL @dkayiwa @rishabhrawat05
